src/graphapi/reduction.cpp

```diff
@@ -134,6 +136,65 @@ void BackendReductionDescriptor::getAttribute(miopenBackendAttributeName_t attri
     }
 }
 
+std::vector<Tensor*> OperationReduction::getInTensors() const { return {mX}; }
+
+std::vector<Tensor*> OperationReduction::getOutTensors() const { return {mY}; }
+
+namespace {
+
+template <typename Ptr>
+void assignPtr(Ptr src, Ptr& dst)
+{
+    if(src != nullptr)
+    {
+        dst = src;
+    }
+    else
+    {
+        MIOPEN_THROW(miopenStatusBadParm);
+    }
+}
```

Comment:
Seems this function is widely used. Can it be moved to a common header?

Reply:
Now it is used in 2 cpp files. I'll propagate its usage into existing cpps and place it into graphapi.hpp in one of the next PRs.

Reply:
Also instead this function we can write:
```C++
dst = &deref(src);
```
I find it less readable. What do you think?

Reply:
For me `defer` doesn't look like a perfect solution too. Current one looks good.

Or
```C++
dst = checkPtr(src);
```

Reply:
Your suggestion is more flexible:
```C++
dst = checkPtr(src);
```
Done so in #2886

Reply:
Done here as well